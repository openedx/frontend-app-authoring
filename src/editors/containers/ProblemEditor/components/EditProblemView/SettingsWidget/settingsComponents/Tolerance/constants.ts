import messages from './messages';

export const ToleranceTypes = {
  percent: {
    type: 'Percent',
    message: messages.typesPercentage,
  },
  number: {
    type: 'Number',
    message: messages.typesNumber,
  },
  none: {
    type: 'None',
    message: messages.typesNone,
  },
} as const;
