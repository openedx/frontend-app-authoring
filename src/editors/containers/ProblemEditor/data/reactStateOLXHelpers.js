import { flatten } from 'lodash';

/**
 * flattenSubNodes - appends a nodes children to a given array 'subNodes' in a flattened format.
 * Only appends direct children unless `recursive` is set to true.
 * @param {object} node
 * @param {object[]} subNodes
 * @param {object} options - { recursive: false } Whether to search recursively for child tags.
 * @returns {void}
 */
function flattenSubNodes(node, subNodes, options = { recursive: false }) {
  const values = Array.isArray(node) ? (
    flatten(node.map(n => Object.values(n)))
  ) : Object.values(node);

  values.forEach(value => {
    if (Array.isArray(value)) {
      subNodes.push(...value);

      if (options.recursive) {
        value.forEach(nestedNode => {
          flattenSubNodes(nestedNode, subNodes, options);
        });
      }
    }
  });
}

/**
 * function nodeContainsChildTags - checks whether a node contains any subnodes with the specified tag names.
 *
 * @param {object} node - Example for node:
 *   {"div":
 *     [
 *       {"label":
 *         [
 *           {"#text":"def"}
 *         ]
 *       },
 *       {"div":
 *         [{
 *           label:
 *             [
 *               { '#text': 'def' },
 *             ],
 *         }],
 *       },
 *       {"#text":" "},
 *       {"em":
 *         [
 *           {"#text":"ghi"}
 *         ],
 *         ":@": {"@_class":"olx_description"}
 *       },
 *       {"em":
 *         [
 *           {"#text":"jkl"}
 *         ]
 *       }
 *     ]
 *   }
 * @param {string[]} tagNames
 * @param {boolean} [recursive=false] - Whether to search recursively for child tags.
 *
 * @returns {boolean} whether node contains the specified child tags
 */
export function nodeContainsChildTags(node, tagNames, options = { recursive: false }) {
  const subNodes = [];
  flattenSubNodes(node, subNodes, options);
  const res = subNodes.some(subNode => tagNames.includes(Object.keys(subNode)[0]));
  return res;
}

/**
 * @param {object} node
 * @returns {string} the tag name of the node
 */
export function tagName(node) {
  if (Array.isArray(node)) {
    throw new TypeError('function tagName does not accept arrays as input');
  }
  return Object.keys(node).find(key => key.match(/^[a-zA-Z].*/));
}

/**
 * findNodesAndRemoveTheirParentNodes - given 'arrayOfNodes', an array that results from the
 * XMLParser, this looks for 'nodesToFind': specific child tags (e.g. 'label'), and removes
 * their parent tags (e.g. 'div') but keeps all their content one level higher.
 * The 'parentsToRemove' array should contain a list of parent tags that will be removed.
 *
 * For example, if you have a div containing a p that contains a label and an em, and you specify
 * 'label' as the nodeToFind and 'p' as the parentToRemove, then you will get a div containing a label and an em.
 * @param {object} { arrayOfNodes, nodesToFind, parentsToRemove }
 *
 * @returns {object} - an array of nodes
 */
export function findNodesAndRemoveTheirParentNodes({
  arrayOfNodes,
  nodesToFind,
  parentsToRemove,
}) {
  const result = [];
  arrayOfNodes.forEach((node) => {
    const containsRelevantSubnodes = nodeContainsChildTags(node, nodesToFind);

    if (!containsRelevantSubnodes || !parentsToRemove.includes(tagName(node))) {
      result.push(node);
    } else {
      const subNodes = Object.values(node)[0];
      result.push(...subNodes);
    }
  });
  return result;
}
