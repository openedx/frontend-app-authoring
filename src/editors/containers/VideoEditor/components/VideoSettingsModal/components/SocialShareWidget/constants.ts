export const analyticsEvents = {
  socialSharingSettingChanged: 'edx.social.video_sharing_setting.changed',
};
