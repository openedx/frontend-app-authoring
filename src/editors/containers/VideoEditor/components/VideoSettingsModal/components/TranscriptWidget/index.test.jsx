import 'CourseAuthoring/editors/setupEditorTest';
import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';

import { RequestKeys } from '../../../../../../data/constants/requests';

import { formatMessage } from '../../../../../../testUtils';
import { actions, selectors } from '../../../../../../data/redux';
// This 'module' self-import hack enables mocking during tests.
// See src/editors/decisions/0005-internal-editor-testability-decisions.md. The whole approach to how hooks are tested
// should be re-thought and cleaned up to avoid this pattern.
// eslint-disable-next-line import/no-self-import
import * as module from './index';

const TranscriptWidget = module.TranscriptWidgetInternal;

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useContext: jest.fn(() => ({ transcripts: ['error.transcripts', jest.fn().mockName('error.setTranscripts')] })),
}));

jest.mock('../../../../../../data/redux', () => ({
  actions: {
    video: {
      updateField: jest.fn().mockName('actions.video.updateField'),
    },
  },
  thunkActions: {
    video: {
      deleteTranscript: jest.fn().mockName('thunkActions.video.deleteTranscript'),
      updateTranscriptHandlerUrl: jest.fn().mockName('thunkActions.video.updateTranscriptHandlerUrl'),
    },
  },

  selectors: {
    app: {
      isLibrary: jest.fn(state => ({ isLibrary: state })),
    },
    video: {
      transcripts: jest.fn(state => ({ transcripts: state })),
      selectedVideoTranscriptUrls: jest.fn(state => ({ selectedVideoTranscriptUrls: state })),
      allowTranscriptDownloads: jest.fn(state => ({ allowTranscriptDownloads: state })),
      showTranscriptByDefault: jest.fn(state => ({ showTranscriptByDefault: state })),
      allowTranscriptImport: jest.fn(state => ({ allowTranscriptImport: state })),
    },
    requests: {
      isFailed: jest.fn(state => ({ isFailed: state })),
    },
  },
}));
jest.mock('../CollapsibleFormWidget', () => 'CollapsibleFormWidget');
jest.mock('./Transcript', () => 'Transcript');

describe('TranscriptWidget', () => {
  describe('hooks', () => {
    describe('transcriptLanguages', () => {
      test('empty list of transcripts returns ', () => {
        expect(module.hooks.transcriptLanguages([])).toEqual('None');
      });
      test('unset gives none', () => {
        expect(module.hooks.transcriptLanguages(['', ''])).toEqual('');
      });
      test('en gives English', () => {
        expect(module.hooks.transcriptLanguages(['en'])).toEqual('English');
      });
      test('en, FR gives English, French', () => {
        expect(module.hooks.transcriptLanguages(['en', 'fr'])).toEqual('English, French');
      });
    });
    describe('hasTranscripts', () => {
      test('null returns false ', () => {
        expect(module.hooks.hasTranscripts(null)).toEqual(false);
      });
      test('empty list returns false', () => {
        expect(module.hooks.hasTranscripts([])).toEqual(false);
      });
      test('content returns true', () => {
        expect(module.hooks.hasTranscripts(['en'])).toEqual(true);
      });
    });
    describe('onAddNewTranscript', () => {
      const mockUpdateField = jest.fn();
      test('null returns [empty string] ', () => {
        module.hooks.onAddNewTranscript({ transcripts: null, updateField: mockUpdateField });
        expect(mockUpdateField).toHaveBeenCalledWith({ transcripts: [''] });
      });
      test(' transcripts return list with blank added', () => {
        const mocklist = ['en', 'fr', 3];
        module.hooks.onAddNewTranscript({ transcripts: mocklist, updateField: mockUpdateField });

        expect(mockUpdateField).toHaveBeenCalledWith({ transcripts: ['en', 'fr', 3, ''] });
      });
    });
  });

  describe('component', () => {
    const props = {
      error: {},
      subtitle: 'SuBTItle',
      title: 'tiTLE',
      intl: { formatMessage },
      transcripts: [],
      selectedVideoTranscriptUrls: {},
      allowTranscriptDownloads: false,
      showTranscriptByDefault: false,
      allowTranscriptImport: false,
      updateField: jest.fn().mockName('args.updateField'),
      isUploadError: false,
      isDeleteError: false,
    };

    describe('snapshots', () => {
      test('snapshots: renders as expected with default props', () => {
        expect(
          shallow(<TranscriptWidget {...props} />).snapshot,
        ).toMatchSnapshot();
      });
      test('snapshots: renders as expected with allowTranscriptImport true', () => {
        expect(
          shallow(<TranscriptWidget {...props} allowTranscriptImport />).snapshot,
        ).toMatchSnapshot();
      });
      test('snapshots: renders as expected with transcripts', () => {
        expect(
          shallow(<TranscriptWidget {...props} transcripts={['en']} />).snapshot,
        ).toMatchSnapshot();
      });
      test('snapshots: renders as expected with transcript urls', () => {
        expect(
          shallow(<TranscriptWidget {...props} selectedVideoTranscriptUrls={{ en: 'url' }} />).snapshot,
        ).toMatchSnapshot();
      });
      test('snapshots: renders as expected with transcripts and urls', () => {
        expect(
          shallow(<TranscriptWidget {...props} transcripts={['es']} selectedVideoTranscriptUrls={{ en: 'url' }} />).snapshot,
        ).toMatchSnapshot();
      });
      test('snapshots: renders as expected with allowTranscriptDownloads true', () => {
        expect(
          shallow(<TranscriptWidget {...props} allowTranscriptDownloads transcripts={['en']} />).snapshot,
        ).toMatchSnapshot();
      });
      test('snapshots: renders as expected with showTranscriptByDefault true', () => {
        expect(
          shallow(<TranscriptWidget {...props} showTranscriptByDefault transcripts={['en']} />).snapshot,
        ).toMatchSnapshot();
      });
      test('snapshot: renders ErrorAlert with upload error message', () => {
        expect(
          shallow(<TranscriptWidget {...props} isUploadError transcripts={['en', 'fr']} />).snapshot,
        ).toMatchSnapshot();
      });
      test('snapshot: renders ErrorAlert with delete error message', () => {
        expect(
          shallow(<TranscriptWidget {...props} isDeleteError transcripts={['en']} />).snapshot,
        ).toMatchSnapshot();
      });
    });
    describe('mapStateToProps', () => {
      const testState = { A: 'pple', B: 'anana', C: 'ucumber' };
      test('transcripts from video.transcript', () => {
        expect(
          module.mapStateToProps(testState).transcripts,
        ).toEqual(selectors.video.transcripts(testState));
      });
      test('allowTranscriptDownloads from video.allowTranscriptDownloads', () => {
        expect(
          module.mapStateToProps(testState).allowTranscriptDownloads,
        ).toEqual(selectors.video.allowTranscriptDownloads(testState));
      });
      test('showTranscriptByDefault from video.showTranscriptByDefault', () => {
        expect(
          module.mapStateToProps(testState).showTranscriptByDefault,
        ).toEqual(selectors.video.showTranscriptByDefault(testState));
      });
      test('allowTranscriptImport from video.allowTranscriptImport', () => {
        expect(
          module.mapStateToProps(testState).allowTranscriptImport,
        ).toEqual(selectors.video.allowTranscriptImport(testState));
      });
      test('isUploadError from requests.isFinished', () => {
        expect(
          module.mapStateToProps(testState).isUploadError,
        ).toEqual(selectors.requests.isFailed(testState, { requestKey: RequestKeys.uploadTranscript }));
      });
      test('isDeleteError from requests.isFinished', () => {
        expect(
          module.mapStateToProps(testState).isDeleteError,
        ).toEqual(selectors.requests.isFailed(testState, { requestKey: RequestKeys.deleteTranscript }));
      });
    });
    describe('mapDispatchToProps', () => {
      const dispatch = jest.fn();
      test('updateField from actions.video.updateField', () => {
        expect(module.mapDispatchToProps.updateField).toEqual(dispatch(actions.video.updateField));
      });
    });
  });
});
