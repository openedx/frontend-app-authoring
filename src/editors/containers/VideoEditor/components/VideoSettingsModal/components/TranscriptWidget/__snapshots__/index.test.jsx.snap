// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TranscriptWidget component snapshots snapshot: renders ErrorAlert with delete error message 1`] = `
<CollapsibleFormWidget
  fontSize="x-small"
  isError={true}
  subtitle="English"
  title="Transcripts"
>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to upload transcript. Please try again."
      description="Message presented to user when transcript fails to upload"
      id="authoring.videoeditor.transcript.error.uploadTranscriptError"
    />
  </ErrorAlert>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={true}
  >
    <FormattedMessage
      defaultMessage="Failed to delete transcript. Please try again."
      description="Message presented to user when transcript fails to delete"
      id="authoring.videoeditor.transcript.error.deleteTranscriptError"
    />
  </ErrorAlert>
  <Stack
    gap={3}
  >
    <Form.Group
      className="border-primary-100 border-bottom"
    >
      <Transcript
        index={0}
        language="en"
      />
      <ActionRow
        className="mt-3.5"
      >
        <Form.Checkbox
          checked={false}
          className="decorative-control-label"
          onChange={[Function]}
        >
          <div
            className="small text-gray-700"
          >
            <FormattedMessage
              defaultMessage="Allow transcript downloads"
              description="Label for allow transcript downloads checkbox"
              id="authoring.videoeditor.transcripts.allowDownloadCheckboxLabel"
            />
          </div>
        </Form.Checkbox>
        <OverlayTrigger
          key="top"
          overlay={
            <Tooltip
              id="tooltip-top"
            >
              <FormattedMessage
                defaultMessage="Learners will see a link to download the transcript below the video."
                description="Message for show by default checkbox"
                id="authoring.videoeditor.transcripts.upload.allowDownloadTooltipMessage"
              />
            </Tooltip>
          }
          placement="top"
        >
          <Icon
            style={
              {
                "height": "16px",
                "width": "16px",
              }
            }
          />
        </OverlayTrigger>
        <ActionRow.Spacer />
      </ActionRow>
      <Form.Checkbox
        checked={false}
        className="mt-3 decorative-control-label"
        onChange={[Function]}
      >
        <div
          className="small text-gray-700"
        >
          <FormattedMessage
            defaultMessage="Show transcript in the video player by default"
            description="Label for show by default checkbox"
            id="authoring.videoeditor.transcripts.upload.showByDefaultCheckboxLabel"
          />
        </div>
      </Form.Checkbox>
    </Form.Group>
    <div
      className="mt-2"
    >
      <Button
        className="text-primary-500 font-weight-bold justify-content-start pl-0"
        onClick={[Function]}
        size="sm"
        variant="link"
      >
        <FormattedMessage
          defaultMessage="Add a transcript"
          description="Label for upload button"
          id="authoring.videoeditor.transcripts.upload.label"
        />
      </Button>
    </div>
  </Stack>
</CollapsibleFormWidget>
`;

exports[`TranscriptWidget component snapshots snapshot: renders ErrorAlert with upload error message 1`] = `
<CollapsibleFormWidget
  fontSize="x-small"
  isError={true}
  subtitle="English, French"
  title="Transcripts"
>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={true}
  >
    <FormattedMessage
      defaultMessage="Failed to upload transcript. Please try again."
      description="Message presented to user when transcript fails to upload"
      id="authoring.videoeditor.transcript.error.uploadTranscriptError"
    />
  </ErrorAlert>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to delete transcript. Please try again."
      description="Message presented to user when transcript fails to delete"
      id="authoring.videoeditor.transcript.error.deleteTranscriptError"
    />
  </ErrorAlert>
  <Stack
    gap={3}
  >
    <Form.Group
      className="border-primary-100 border-bottom"
    >
      <Transcript
        index={0}
        language="en"
      />
      <Transcript
        index={1}
        language="fr"
      />
      <ActionRow
        className="mt-3.5"
      >
        <Form.Checkbox
          checked={false}
          className="decorative-control-label"
          onChange={[Function]}
        >
          <div
            className="small text-gray-700"
          >
            <FormattedMessage
              defaultMessage="Allow transcript downloads"
              description="Label for allow transcript downloads checkbox"
              id="authoring.videoeditor.transcripts.allowDownloadCheckboxLabel"
            />
          </div>
        </Form.Checkbox>
        <OverlayTrigger
          key="top"
          overlay={
            <Tooltip
              id="tooltip-top"
            >
              <FormattedMessage
                defaultMessage="Learners will see a link to download the transcript below the video."
                description="Message for show by default checkbox"
                id="authoring.videoeditor.transcripts.upload.allowDownloadTooltipMessage"
              />
            </Tooltip>
          }
          placement="top"
        >
          <Icon
            style={
              {
                "height": "16px",
                "width": "16px",
              }
            }
          />
        </OverlayTrigger>
        <ActionRow.Spacer />
      </ActionRow>
      <Form.Checkbox
        checked={false}
        className="mt-3 decorative-control-label"
        onChange={[Function]}
      >
        <div
          className="small text-gray-700"
        >
          <FormattedMessage
            defaultMessage="Show transcript in the video player by default"
            description="Label for show by default checkbox"
            id="authoring.videoeditor.transcripts.upload.showByDefaultCheckboxLabel"
          />
        </div>
      </Form.Checkbox>
    </Form.Group>
    <div
      className="mt-2"
    >
      <Button
        className="text-primary-500 font-weight-bold justify-content-start pl-0"
        onClick={[Function]}
        size="sm"
        variant="link"
      >
        <FormattedMessage
          defaultMessage="Add a transcript"
          description="Label for upload button"
          id="authoring.videoeditor.transcripts.upload.label"
        />
      </Button>
    </div>
  </Stack>
</CollapsibleFormWidget>
`;

exports[`TranscriptWidget component snapshots snapshots: renders as expected with allowTranscriptDownloads true 1`] = `
<CollapsibleFormWidget
  fontSize="x-small"
  isError={true}
  subtitle="English"
  title="Transcripts"
>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to upload transcript. Please try again."
      description="Message presented to user when transcript fails to upload"
      id="authoring.videoeditor.transcript.error.uploadTranscriptError"
    />
  </ErrorAlert>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to delete transcript. Please try again."
      description="Message presented to user when transcript fails to delete"
      id="authoring.videoeditor.transcript.error.deleteTranscriptError"
    />
  </ErrorAlert>
  <Stack
    gap={3}
  >
    <Form.Group
      className="border-primary-100 border-bottom"
    >
      <Transcript
        index={0}
        language="en"
      />
      <ActionRow
        className="mt-3.5"
      >
        <Form.Checkbox
          checked={true}
          className="decorative-control-label"
          onChange={[Function]}
        >
          <div
            className="small text-gray-700"
          >
            <FormattedMessage
              defaultMessage="Allow transcript downloads"
              description="Label for allow transcript downloads checkbox"
              id="authoring.videoeditor.transcripts.allowDownloadCheckboxLabel"
            />
          </div>
        </Form.Checkbox>
        <OverlayTrigger
          key="top"
          overlay={
            <Tooltip
              id="tooltip-top"
            >
              <FormattedMessage
                defaultMessage="Learners will see a link to download the transcript below the video."
                description="Message for show by default checkbox"
                id="authoring.videoeditor.transcripts.upload.allowDownloadTooltipMessage"
              />
            </Tooltip>
          }
          placement="top"
        >
          <Icon
            style={
              {
                "height": "16px",
                "width": "16px",
              }
            }
          />
        </OverlayTrigger>
        <ActionRow.Spacer />
      </ActionRow>
      <Form.Checkbox
        checked={false}
        className="mt-3 decorative-control-label"
        onChange={[Function]}
      >
        <div
          className="small text-gray-700"
        >
          <FormattedMessage
            defaultMessage="Show transcript in the video player by default"
            description="Label for show by default checkbox"
            id="authoring.videoeditor.transcripts.upload.showByDefaultCheckboxLabel"
          />
        </div>
      </Form.Checkbox>
    </Form.Group>
    <div
      className="mt-2"
    >
      <Button
        className="text-primary-500 font-weight-bold justify-content-start pl-0"
        onClick={[Function]}
        size="sm"
        variant="link"
      >
        <FormattedMessage
          defaultMessage="Add a transcript"
          description="Label for upload button"
          id="authoring.videoeditor.transcripts.upload.label"
        />
      </Button>
    </div>
  </Stack>
</CollapsibleFormWidget>
`;

exports[`TranscriptWidget component snapshots snapshots: renders as expected with allowTranscriptImport true 1`] = `
<CollapsibleFormWidget
  fontSize="x-small"
  isError={true}
  subtitle="None"
  title="Transcripts"
>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to upload transcript. Please try again."
      description="Message presented to user when transcript fails to upload"
      id="authoring.videoeditor.transcript.error.uploadTranscriptError"
    />
  </ErrorAlert>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to delete transcript. Please try again."
      description="Message presented to user when transcript fails to delete"
      id="authoring.videoeditor.transcript.error.deleteTranscriptError"
    />
  </ErrorAlert>
  <Stack
    gap={3}
  >
    <Fragment>
      <FormattedMessage
        defaultMessage="Add video transcripts (.srt files only) for improved accessibility."
        description="Message for adding first transcript"
        id="authoring.videoeditor.transcripts.upload.firstTranscriptMessage"
      />
      <injectIntl(ShimmedIntlComponent)
        setOpen={[Function]}
      />
    </Fragment>
    <div
      className="mt-2"
    >
      <Button
        className="text-primary-500 font-weight-bold justify-content-start pl-0"
        onClick={[Function]}
        size="sm"
        variant="link"
      >
        <FormattedMessage
          defaultMessage="Add a transcript"
          description="Label for upload button"
          id="authoring.videoeditor.transcripts.upload.label"
        />
      </Button>
    </div>
  </Stack>
</CollapsibleFormWidget>
`;

exports[`TranscriptWidget component snapshots snapshots: renders as expected with default props 1`] = `
<CollapsibleFormWidget
  fontSize="x-small"
  isError={true}
  subtitle="None"
  title="Transcripts"
>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to upload transcript. Please try again."
      description="Message presented to user when transcript fails to upload"
      id="authoring.videoeditor.transcript.error.uploadTranscriptError"
    />
  </ErrorAlert>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to delete transcript. Please try again."
      description="Message presented to user when transcript fails to delete"
      id="authoring.videoeditor.transcript.error.deleteTranscriptError"
    />
  </ErrorAlert>
  <Stack
    gap={3}
  >
    <Fragment>
      <FormattedMessage
        defaultMessage="Add video transcripts (.srt files only) for improved accessibility."
        description="Message for adding first transcript"
        id="authoring.videoeditor.transcripts.upload.firstTranscriptMessage"
      />
    </Fragment>
    <div
      className="mt-2"
    >
      <Button
        className="text-primary-500 font-weight-bold justify-content-start pl-0"
        onClick={[Function]}
        size="sm"
        variant="link"
      >
        <FormattedMessage
          defaultMessage="Add a transcript"
          description="Label for upload button"
          id="authoring.videoeditor.transcripts.upload.label"
        />
      </Button>
    </div>
  </Stack>
</CollapsibleFormWidget>
`;

exports[`TranscriptWidget component snapshots snapshots: renders as expected with showTranscriptByDefault true 1`] = `
<CollapsibleFormWidget
  fontSize="x-small"
  isError={true}
  subtitle="English"
  title="Transcripts"
>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to upload transcript. Please try again."
      description="Message presented to user when transcript fails to upload"
      id="authoring.videoeditor.transcript.error.uploadTranscriptError"
    />
  </ErrorAlert>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to delete transcript. Please try again."
      description="Message presented to user when transcript fails to delete"
      id="authoring.videoeditor.transcript.error.deleteTranscriptError"
    />
  </ErrorAlert>
  <Stack
    gap={3}
  >
    <Form.Group
      className="border-primary-100 border-bottom"
    >
      <Transcript
        index={0}
        language="en"
      />
      <ActionRow
        className="mt-3.5"
      >
        <Form.Checkbox
          checked={false}
          className="decorative-control-label"
          onChange={[Function]}
        >
          <div
            className="small text-gray-700"
          >
            <FormattedMessage
              defaultMessage="Allow transcript downloads"
              description="Label for allow transcript downloads checkbox"
              id="authoring.videoeditor.transcripts.allowDownloadCheckboxLabel"
            />
          </div>
        </Form.Checkbox>
        <OverlayTrigger
          key="top"
          overlay={
            <Tooltip
              id="tooltip-top"
            >
              <FormattedMessage
                defaultMessage="Learners will see a link to download the transcript below the video."
                description="Message for show by default checkbox"
                id="authoring.videoeditor.transcripts.upload.allowDownloadTooltipMessage"
              />
            </Tooltip>
          }
          placement="top"
        >
          <Icon
            style={
              {
                "height": "16px",
                "width": "16px",
              }
            }
          />
        </OverlayTrigger>
        <ActionRow.Spacer />
      </ActionRow>
      <Form.Checkbox
        checked={true}
        className="mt-3 decorative-control-label"
        onChange={[Function]}
      >
        <div
          className="small text-gray-700"
        >
          <FormattedMessage
            defaultMessage="Show transcript in the video player by default"
            description="Label for show by default checkbox"
            id="authoring.videoeditor.transcripts.upload.showByDefaultCheckboxLabel"
          />
        </div>
      </Form.Checkbox>
    </Form.Group>
    <div
      className="mt-2"
    >
      <Button
        className="text-primary-500 font-weight-bold justify-content-start pl-0"
        onClick={[Function]}
        size="sm"
        variant="link"
      >
        <FormattedMessage
          defaultMessage="Add a transcript"
          description="Label for upload button"
          id="authoring.videoeditor.transcripts.upload.label"
        />
      </Button>
    </div>
  </Stack>
</CollapsibleFormWidget>
`;

exports[`TranscriptWidget component snapshots snapshots: renders as expected with transcript urls 1`] = `
<CollapsibleFormWidget
  fontSize="x-small"
  isError={true}
  subtitle="None"
  title="Transcripts"
>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to upload transcript. Please try again."
      description="Message presented to user when transcript fails to upload"
      id="authoring.videoeditor.transcript.error.uploadTranscriptError"
    />
  </ErrorAlert>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to delete transcript. Please try again."
      description="Message presented to user when transcript fails to delete"
      id="authoring.videoeditor.transcript.error.deleteTranscriptError"
    />
  </ErrorAlert>
  <Stack
    gap={3}
  >
    <Fragment>
      <FormattedMessage
        defaultMessage="Add video transcripts (.srt files only) for improved accessibility."
        description="Message for adding first transcript"
        id="authoring.videoeditor.transcripts.upload.firstTranscriptMessage"
      />
    </Fragment>
    <div
      className="mt-2"
    >
      <Button
        className="text-primary-500 font-weight-bold justify-content-start pl-0"
        onClick={[Function]}
        size="sm"
        variant="link"
      >
        <FormattedMessage
          defaultMessage="Add a transcript"
          description="Label for upload button"
          id="authoring.videoeditor.transcripts.upload.label"
        />
      </Button>
    </div>
  </Stack>
</CollapsibleFormWidget>
`;

exports[`TranscriptWidget component snapshots snapshots: renders as expected with transcripts 1`] = `
<CollapsibleFormWidget
  fontSize="x-small"
  isError={true}
  subtitle="English"
  title="Transcripts"
>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to upload transcript. Please try again."
      description="Message presented to user when transcript fails to upload"
      id="authoring.videoeditor.transcript.error.uploadTranscriptError"
    />
  </ErrorAlert>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to delete transcript. Please try again."
      description="Message presented to user when transcript fails to delete"
      id="authoring.videoeditor.transcript.error.deleteTranscriptError"
    />
  </ErrorAlert>
  <Stack
    gap={3}
  >
    <Form.Group
      className="border-primary-100 border-bottom"
    >
      <Transcript
        index={0}
        language="en"
      />
      <ActionRow
        className="mt-3.5"
      >
        <Form.Checkbox
          checked={false}
          className="decorative-control-label"
          onChange={[Function]}
        >
          <div
            className="small text-gray-700"
          >
            <FormattedMessage
              defaultMessage="Allow transcript downloads"
              description="Label for allow transcript downloads checkbox"
              id="authoring.videoeditor.transcripts.allowDownloadCheckboxLabel"
            />
          </div>
        </Form.Checkbox>
        <OverlayTrigger
          key="top"
          overlay={
            <Tooltip
              id="tooltip-top"
            >
              <FormattedMessage
                defaultMessage="Learners will see a link to download the transcript below the video."
                description="Message for show by default checkbox"
                id="authoring.videoeditor.transcripts.upload.allowDownloadTooltipMessage"
              />
            </Tooltip>
          }
          placement="top"
        >
          <Icon
            style={
              {
                "height": "16px",
                "width": "16px",
              }
            }
          />
        </OverlayTrigger>
        <ActionRow.Spacer />
      </ActionRow>
      <Form.Checkbox
        checked={false}
        className="mt-3 decorative-control-label"
        onChange={[Function]}
      >
        <div
          className="small text-gray-700"
        >
          <FormattedMessage
            defaultMessage="Show transcript in the video player by default"
            description="Label for show by default checkbox"
            id="authoring.videoeditor.transcripts.upload.showByDefaultCheckboxLabel"
          />
        </div>
      </Form.Checkbox>
    </Form.Group>
    <div
      className="mt-2"
    >
      <Button
        className="text-primary-500 font-weight-bold justify-content-start pl-0"
        onClick={[Function]}
        size="sm"
        variant="link"
      >
        <FormattedMessage
          defaultMessage="Add a transcript"
          description="Label for upload button"
          id="authoring.videoeditor.transcripts.upload.label"
        />
      </Button>
    </div>
  </Stack>
</CollapsibleFormWidget>
`;

exports[`TranscriptWidget component snapshots snapshots: renders as expected with transcripts and urls 1`] = `
<CollapsibleFormWidget
  fontSize="x-small"
  isError={true}
  subtitle="Spanish"
  title="Transcripts"
>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to upload transcript. Please try again."
      description="Message presented to user when transcript fails to upload"
      id="authoring.videoeditor.transcript.error.uploadTranscriptError"
    />
  </ErrorAlert>
  <ErrorAlert
    dismissError={null}
    hideHeading={true}
    isError={false}
  >
    <FormattedMessage
      defaultMessage="Failed to delete transcript. Please try again."
      description="Message presented to user when transcript fails to delete"
      id="authoring.videoeditor.transcript.error.deleteTranscriptError"
    />
  </ErrorAlert>
  <Stack
    gap={3}
  >
    <Form.Group
      className="border-primary-100 border-bottom"
    >
      <Transcript
        index={0}
        language="es"
      />
      <ActionRow
        className="mt-3.5"
      >
        <Form.Checkbox
          checked={false}
          className="decorative-control-label"
          onChange={[Function]}
        >
          <div
            className="small text-gray-700"
          >
            <FormattedMessage
              defaultMessage="Allow transcript downloads"
              description="Label for allow transcript downloads checkbox"
              id="authoring.videoeditor.transcripts.allowDownloadCheckboxLabel"
            />
          </div>
        </Form.Checkbox>
        <OverlayTrigger
          key="top"
          overlay={
            <Tooltip
              id="tooltip-top"
            >
              <FormattedMessage
                defaultMessage="Learners will see a link to download the transcript below the video."
                description="Message for show by default checkbox"
                id="authoring.videoeditor.transcripts.upload.allowDownloadTooltipMessage"
              />
            </Tooltip>
          }
          placement="top"
        >
          <Icon
            style={
              {
                "height": "16px",
                "width": "16px",
              }
            }
          />
        </OverlayTrigger>
        <ActionRow.Spacer />
      </ActionRow>
      <Form.Checkbox
        checked={false}
        className="mt-3 decorative-control-label"
        onChange={[Function]}
      >
        <div
          className="small text-gray-700"
        >
          <FormattedMessage
            defaultMessage="Show transcript in the video player by default"
            description="Label for show by default checkbox"
            id="authoring.videoeditor.transcripts.upload.showByDefaultCheckboxLabel"
          />
        </div>
      </Form.Checkbox>
    </Form.Group>
    <div
      className="mt-2"
    >
      <Button
        className="text-primary-500 font-weight-bold justify-content-start pl-0"
        onClick={[Function]}
        size="sm"
        variant="link"
      >
        <FormattedMessage
          defaultMessage="Add a transcript"
          description="Label for upload button"
          id="authoring.videoeditor.transcripts.upload.label"
        />
      </Button>
    </div>
  </Stack>
</CollapsibleFormWidget>
`;
